File: 2271-rearrange-array-elements-by-sign/rearrange-array-elements-by-sign.cpp
class Solution {
public:
    vector<int> rearrangeArray(vector<int>& nums) {
        int pos=0;
        int neg = 1;
        vector<int>result(nums.size());
        for(int num:nums){
            if(num>0){
                result[pos]= num;
                pos +=2;

            }

            if(num<0){
                result[neg]=num;
                neg +=2;

            }
        }

        return result;
    }
};

File: 2271-rearrange-array-elements-by-sign/rearrange-array-elements-by-sign.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var rearrangeArray = function(nums) {
    let negarr =[];
    let posarr =[];

    let result =[];

    for(let num of nums){
        if (num<0){
            negarr.push(num);
        }
    }

    for(let num of nums){
        if(num>0){
            posarr.push(num);
        }
    }

    for(let i=0;i<posarr.length;i++){
        result.push(posarr[i]);
        result.push(negarr[i]);
    }

    return result;
};